pkg/targetallocator/adapters/config_to_prom_config.go

```diff
@@ -57,3 +68,208 @@ func ConfigToPromConfig(cfg string) (map[interface{}]interface{}, error) {
 
 	return prometheus, nil
 }
+
+// UnescapeDollarSignsInPromConfig replaces "$$" with "$" in the "replacement" fields of
+// both "relabel_configs" and "metric_relabel_configs" in a Prometheus configuration file.
+func UnescapeDollarSignsInPromConfig(cfg string) (map[interface{}]interface{}, error) {
```

Comment:
What is the test coverage like on these new functions? It seems like there are a lot of `if` statements present but very few test cases.

Reply:
Adding test cases for missing and invalid configurations as well.